### DISCUSSION TITLE
should CONTRIBUTING.md suggest use of virtualenvs/poetry?

### DISCUSSION THREAD

**Discussion by petergaultney** [zauberzeug/nicegui#2710] [Ideas / Feature Requests]
I see from your CONTRIBUTING.md that it may be standard to do nicegui development inside a docker container. Some of us (😄 ) don't really use VSCode and would prefer not to use Docker where we can avoid it, too.

In my experience, it's somewhat standard in the Python community to do development inside virtual environments. And in fact you already use poetry, define all your dev dependencies there, etc.

Should we change CONTRIBUTING.md to suggest that contributors install poetry and use `poetry install --sync --all-extras` to get their venv set up for development? That seems quite a bit simpler than the global install approach that's suggested right now, with `python -m pip install -e .` etc.

**Comment by kleynjan** [comment]
@petergaultney +1 on this. The Docker suggestion/requirement initially discouraged me from setting up a NiceGUI dev env. And I agree that nobody should be doing global installs anymore. I am still using conda but perhaps it's time to switch to poetry anyway.

**Comment by rodja** [comment]
Sure @petergaultney, we would gladly incorporate such a pull request. Should be not to hard as @intrinsically just stated in https://github.com/zauberzeug/nicegui/issues/2611#issuecomment-2021709826. I guess the docker container will still be simpler if you want to run the tests. But for a quick start venv/poetry is quite nice.
